openedx/openedx-platform
# refactor: Simplify ModuleStore Runtimes now that XModules are gone

## Description

```
* Consolidates and renames the runtime which is used as a base for all the others:
  * Before: `xmodule.x_module:DescriptorSystem` and
            `xmodule.mako_block:MakoDescriptorSystem`.
  * After:  `xmodule.x_module:ModuleStoreRuntime`.

* Co-locates and renames the runtimes for importing course OLX:
  * Before: `xmodule.x_module:XMLParsingSystem` and
            `xmodule.modulestore.xml:ImportSystem`.
  * After:  `xmodule.modulestore.xml:XMLParsingModuleStoreRuntime` and
            `xmodule.modulestore.xml:XMLImportingModuleStoreRuntime`.
  * Note: I would have liked to consolidate these, but it would have
          involved nontrivial test refactoring.

* Renames the stub Old Mongo runtime:
  * Before: `xmodule.modulestore.mongo.base:CachingDescriptorSystem`.
  * After: `xmodule.modulestore.mongo.base:OldModuleStoreRuntime`.

* Renames the Split Mongo runtime, the which is what runs courses in LMS and CMS:
  * Before: `xmodule.modulestore.split_mongo.caching_descriptor_system:CachingDescriptorSystem`.
  * After: `xmodule.modulestore.split_mongo.runtime:SplitModuleStoreRuntime`.

* Renames some of the dummy runtimes used only in unit tests.
```

## Other info

This should have no user or operator impact.

## Testing instructions

Smoke test general LMS and CMS usage on linked sandbox. Confirm that the demo course can be round-tripped through OLX (import, export, re-import), as that uses a different runtime class than typical LMS/CMS rendering.

## Deadline

Before the Ulmo cutoff would be nice, as it would save 2U the trouble of having to handle this in the Ulmo->Verawood. But, it's not critical.
